### ISSUE TITLE
Icon spacing issue after locking/unlocking

### ISSUE THREAD

**Issue by parkerlreed** [jderose9/dash-to-panel#85]
Arch x86_64
gnome-shell-extension-dash-to-panel-git r129.7bba9a4-1

Whether it be because display times out or system goes to sleep, on two machines now (with the same settings) the icon spacing is messed up after unlocking.

Normal

![screenshot from 2017-03-08 19-49-24](https://cloud.githubusercontent.com/assets/841440/23730918/66fe4d82-0438-11e7-993a-e5092ad5890f.png)

After unlocking

![screenshot from 2017-03-08 19-50-24](https://cloud.githubusercontent.com/assets/841440/23730927/7dfdbd10-0438-11e7-9d98-a527f1712fae.png)

Settings 

![screenshot from 2017-03-08 19-51-26](https://cloud.githubusercontent.com/assets/841440/23730958/a98ff952-0438-11e7-85f4-aede9c53a74e.png)
![screenshot from 2017-03-08 19-51-17](https://cloud.githubusercontent.com/assets/841440/23730956/a98d84b0-0438-11e7-9a50-46b3fdf16650.png)
![screenshot from 2017-03-08 19-51-04](https://cloud.githubusercontent.com/assets/841440/23730957/a98e4512-0438-11e7-8342-92bc2733a14c.png)

**Comment by jderose9** [comment]
I definitely haven't seen this. I'm guessing it is not loading your setting from dconf for some reason and instead going back to the default of 8? If you set it to 8 does it stay the same size? Are you on Gnome 3.22 or 3.24? Thanks

**Comment by parkerlreed** [comment]
@jderose9 Not quite. It seem the abnormal state adds extra padding to every size level. https://my.mixtape.moe/grkfik.mp4

**Comment by jderose9** [comment]
So at 00:10 you are locking and unlocking your screen?

**Comment by parkerlreed** [comment]
@jderose9 Yep

**Comment by jderose9** [comment]
Can you try to disable every other extension to verify that the issue isn't caused/compounded by another extension?

**Comment by parkerlreed** [comment]
Aha! Yes, the only other extension I have is causing it somehow.

https://github.com/phocean/TopIcons-plus

**Comment by jderose9** [comment]
Are you sure you have "TopIcons Plus" rather than "TopIcons"?

**Comment by parkerlreed** [comment]
Aye. I install from AUR which pulls from that repo 

https://aur.archlinux.org/packages/gnome-shell-extension-topicons-plus-git/

**Comment by parkerlreed** [comment]
It's still happening but less frequently with the only extension enabled being dash-to-panel

**Comment by jderose9** [comment]
Are you using window scaling for a HiDPI monitor?

Does the spacing keep growing if you lock/unlock multiple times or once it's messed up, it just stays that way?

**Comment by parkerlreed** [comment]
No scaling, all on 1080p. Same spacing each time. It stays normal for a handful of lock cycles but then will go to that wide spacing. Then another cycle and it's back to normal.

**Comment by galfwender** [comment]
I have the same issue

GNOME Shell 3.24.0
DtP v8

**Comment by SolarLiner** [comment]
Same issue here on Ubuntu GNOME 17.04, GNOME 3.24, on a 1366x768 laptop. Seems like after unlocking a padding of 8 is added on top of the padding setting in Dash to Panel.

**Comment by TechnoSparks** [comment]
Same behaviour with OP, although mine happens all the time. I think it would be helpful to add that not only locking/unlocking produces this behaviour, but actually anything that involves redrawing(?) the Gnome Panel.

1. Disabling/enabling User Themes
2. Disabling/enabling “Removable Drive Menu”
3. Disabling/enabling Dash to Panel itself
4. Disabling/enabling Extension master switch

Removing and reinstalling the extension doesn't remedy the issue. I have also tried leaving Dash to Panel enabled alone, and the issue persisted.. It is worth to note that, because Dash to Panel has some parts from Dash to Dock, the latter extension doesn't have this issue. Although my (previously closed because duplicate) issue and this comment is new, I want to make clear that this issue happened for a long time now, but it has gotten annoying since I recently lock/unlock a lot :(

Gnome 3.24.2
Dash to Panel v8
Issue happened since before 3.24 prime release

I am also looking forward to deleting the config files to see if it would help. Could anyone enlighten me on the location of the config files?

_If there are any relations_, I also think that this may related to the reason that new task (when you launch a new window) animation isn't properly executed. I'll try to get video evidence (and also read all open issues lel).

**Comment by jderose9** [comment]
@TechnoSparks There's definitely quite a few issues reported (that I don't observe myself) that seem to be related to drawing on the display that should be occurring that isn't. I don't know the extent to which they are related.

**Comment by mozartilize** [comment]
I experience the same issue too, ubuntu gnome 16.04, gnome-shell 3.20

**Comment by TechnoSparks** [comment]
I tried to clear the configs and changed the settings one by one and after many many trials it seemed there is no definitive way to find the cause.

The only thing in common of all tests is that, any changes from the default settings will contribute to this behaviour.

Leaving default settings will not​ cause the padding to be added. I don't know what would happen after a restart though, didn't check that _yet_ .

Possible solution suggestion: force the extension to check if the padding is correct otherwise refresh

**Comment by TechnoSparks** [comment]
@jderose9 i may not understand fully the (possible related) issue that you are mentioning. 😯 Can you elaborate?

**Comment by TechnoSparks** [comment]
been 4 days since my last post, and after 4 tests of leaving the session locked for 9 hours, the padding is still added to the panel

**Comment by jderose9** [comment]
@TechnoSparks I think your message about missing schema if in dconf editor is simply because that dconf editor application can't find the schema definition included with the extension. That doesn't mean the extension itself can't find and load it so I don't think it's necessarily related to your issue. If you delete all the keys the extension will just use the defaults from the schema.

Gnome disables the extensions when the screen is locked and re-enables when it is unlocked, and it seems like something with that process is just not happening cleanly.

I've switched away from the default app icon margin to help try to reproduce so thanks for that additional info.

**Comment by TechnoSparks** [comment]
By “default settings”, I meant the extension's default settings for everything. Simply changing another unrelated setting (something that is not about the padding) could even add the extra padding to the already default 8px padding.

Could necro or old Gnome settings be the cause of this?

My Gnome desktop uses the same configs that were generated when I was using Ubuntu Gnome (now Manjaro Gnome), thanks to the magic of using the same /home partition.

Time to check that. Resetting everything Gnome now

**Comment by TechnoSparks** [comment]
@jderose9 I managed to squash the issue by running
```
dconf reset -f "/org/gnome/shell/extensions/dash-to-panel/"
```
in a terminal rather than resetting the keys via Dconf Editor itself. I always thought that it could be with the configs, since back then when I reset the keys via Dconf, there is this one key that doesn't want to go away. Sadly since I am now in a fixed state, I cannot get back to my previous broken state.

I would invite other participants of this issue to try the command above and see if it worked, especially the OP @parkerlreed

However, it is also worthy to note that after I did this, I still saw spacings being added, but only to running applications after locking (favourites or not). Ending all tasks of the said application will make the spacing reset to its correct state (favourites), no longer the need to do a shell restart. But shell restart will also fix it. This will not happen if icon size and padding is in their default values. It seems this may be the last issue lingering perhaps?

Wish to hear back from the participants and the developer.

**Comment by parkerlreed** [comment]
@TechnoSparks That seems to have worked. The icons end up the same spacing after a lock/unlock.

**Comment by parkerlreed** [comment]
It's even screwier than before... Fresh reset plus I updated from the repo.

First two dark screens are just an Alt F2 + r (spacing becomes condensed)

Last is a screen lock (spacing becomes wider)

https://www.youtube.com/watch?v=nLdJIwdBoxQ

**Comment by TechnoSparks** [comment]
That is disappointing 🙁

By “repo” did you mean you are running the version 8 version of the extension? @parkerlreed

**Comment by parkerlreed** [comment]
I meant from git, this repo.

**Comment by hubslave** [comment]
@jderose9 I was experimenting a little bit more with cases when does this issue happen and probably found out what is the problem

**1.** It seems that this problem does not happen when users have their machines set up in a way that screensaver also locks their machines - so when they deactivate screensaver they have to insert password and then they get to desktop just fine. Icons are arranged as they should be.

You can activate a screensaver **with lockscreen** by this command:

`dbus-send --type=method_call --dest=org.gnome.ScreenSaver /org/gnome/ScreenSaver org.gnome.ScreenSaver.Lock`

It also seems that screensaver with lockscreen **restarts gnome session** after unlocking, since if I have my icons "broken" as described in this issue, activating screensaver with locking the screen repairs this problem.

**2.** But there is also second case: Users can have their machines set up in a way that screensaver does not lock their computer (I also have my computer set up this way.) So deactivating screensaver automatically gets me to desktop, without having to write password.

You can activate a screensaver **without lockscreen** by this command:

`dbus-send --type=method_call --dest=org.gnome.ScreenSaver /org/gnome/ScreenSaver org.gnome.ScreenSaver.SetActive boolean:true`

In my case after running this command, (sometimes I have to do it few times) breaks the icons spacing as described in this thread.

It seems that screensaver without lockscreen **does not restart gnome session** after unlocking and this can broke the icons spacing as described in this issue.

If you need any more information or testing please let me know

**Comment by parkerlreed** [comment]
Dear lord it seems that way :D (I did a Win L lock a few times (password) and it stayed normal. No password lock did mess it up)

**Comment by TechnoSparks** [comment]
Hello,

I have my setup that it requires my password after unlocking. However, as pointed [here](#issuecomment-314124247)
 in my last post, the icon spacing issue is still present **only for** running tasks. This will happen with either locking manually (I have my shortcut as Super+L) or leaving the machine until timeout.

What is really weird here is that invoking the command `dbus-send --type=method_call --dest=org.gnome.ScreenSaver /org/gnome/ScreenSaver org.gnome.ScreenSaver.Lock` **do not** add the weird padding to the icons.

![screenshot from 2017-09-22 10-58-55](https://user-images.githubusercontent.com/6072210/30727675-4ea2cd2e-9f85-11e7-977c-a688354acdab.png)
This image shows that I have my machine is “set up in a way that screensaver also locks their machines” through the System Settings

![screenshot from 2017-09-22 11-04-40](https://user-images.githubusercontent.com/6072210/30727751-e9371b38-9f85-11e7-80a6-06217d7a70d9.png)
This image shows that I am using the built-in key shortcut feature to manually lock my screen

Does Gnome Desktop call a different command when it or I "lock" the device? This is confusing. As for a workaround, is there any way I can bind the command to my Super+L combo?

**Comment by jderose9** [comment]
I appreciate all of the details, but unfortunately I still can't reproduce this.

When you guys say "Screen Saver" you mean the power option built into Gnome to disable that shuts off the display, correct?

When I execute this command, either via key binding or terminal, the screen still locks.
`dbus-send --type=method_call --dest=org.gnome.ScreenSaver /org/gnome/ScreenSaver org.gnome.ScreenSaver.SetActive boolean:true`

 If I remove the password from my account, I still get the screen shield that has to be dragged/swiped away before the desktop is visible (though I don't have to enter my password after). I am unable to make the screen idle without the shield coming back up when it is waked.

The extensions do all disable and re-enable when the screen is locked/unlocked. I don't think it's necessarily a matter of the commands being different necessarily, but rather executed in a different order. Like, perhaps the custom dconf settings haven't been loaded yet or the default style gets applied after this extensions.

**Comment by hubslave** [comment]
Hello, sorry for late response.

When talking about screen saver I mean - screen turns off and after turning it on I have there "screen shield" - blue screen with time and date which has to be dragged/swiped away before desktop is visible. Yes I cant make screen idle without the shield coming back up too. And also the same with locking - If I disable password I dont need to type it after "removing" screen shield - I go straight to desktop. But in that case dash to panel sometimes breaks up as described here. When I have password enabled and after "removing" screen shield "login screen" appears (where I have to type my password) it seems that gnome sessions restarts in background before coming back to desktop and this issue does not happen.

**Comment by robvelor** [comment]
This happens to me all the time as well since I first started this extension daily a few months back, a quick restart with Alt+F2+r always fixes the issue. Also note that not only locking/unlocking the session causes it but also simple updating other extensions as well, or even a crash on another extension might cause it.

**Comment by v8d** [comment]
One easy way to reproduce this bug: Set the panel size to 24 and the app icon margin to 0, then toggle the extension off/on. It appears that it uses the default app icon padding (8) whenever it's not started via a shell restart. I also noticed the icons jump around while the extension is starting, as if two different padding settings are being applied.

**Comment by robvelor** [comment]
@v8d Indeed I can reproduce this way, but also happens regardless of animations being on or off.

**Comment by cmivxx** [comment]
Thought I would toss in an update.  For me, this only happens on 3.24 when running XORG.  If I run it under Wayland, it works fine.  Hope that helps someone.

**Comment by jderose9** [comment]
Thanks! Turning off gnome's animations was what I needed to do to reproduce on my machine. I believe this is now resolved. Can you guys try the newest master? Thanks!

**Comment by TechnoSparks** [comment]
This is extremely exciting! 😆
I have installed the version from master and have rummaged through many lock shades and so far I can gladly say the issue has been gone!

I do think we have to wait for other participants to test this as well.

Thank you jdrose9!!

**Comment by jderose9** [comment]
@TechnoSparks Looks like it's shifted to the left. This happens after unlocking only?

**Comment by TechnoSparks** [comment]
@jdrose9 It sure looks like so, but it is not. This is because I have 1px margin between items, and in the screenshot, there is 1px line at the left. And yes, it happens only after unlocking.

This does not bother me much, especially because it happens very rarely (I tried to recreate it) that I don't think it would need serious considerations.

**Comment by hubslave** [comment]
I can confirm that this issue is gone after last update of dash-to-panel on my machine. Thank you very much :-)

**Comment by TungstenOxide** [comment]
I'm getting this issue too. I have a workaround of every time it happens, I just set the icon size to 0 and then bring it back up to 30.
![panel1](https://user-images.githubusercontent.com/12565248/35199157-5ef3773a-fec8-11e7-8e51-f2da28c2344f.png)
![panel2](https://user-images.githubusercontent.com/12565248/35199158-5f028ebe-fec8-11e7-9266-cc8a73a39517.png)
![panel3](https://user-images.githubusercontent.com/12565248/35199159-5f11f57a-fec8-11e7-82dd-400525af8b42.png)

**Comment by robvelor** [comment]
@jderose9 I will give it a spin and report back, this was so annoying for me that I went back to the gnome3 paradigm.

**Comment by jderose9** [comment]
Yes, if anyone wants to try latest `master` and report back, it would be great. The work by @charlesg99  may have resolved it. Thanks!

**Comment by TungstenOxide** [comment]
When will this update be on extensions.gnome.org?

**Comment by jderose9** [comment]
It will be a few weeks. Ironing out a couple of issues and then I will submit, but it takes a few weeks for someone to review. I'm not actually sure whether this one is completely resolved though - if someone that was recently experiencing it can try the master branch and let me know, that would be awesome! I'm going to close for now with the assumption that it is resolved, but feel free to post back if you find otherwise.

**Comment by TechnoSparks** [comment]
I have tried the latest branch and it seemed to be fine (although the one before the recent merge also worked fine too). I could also safely say that the little visual glitch I posted earlier in this issue seems to be gone now.

**Comment by jderose9** [comment]
@TechnoSparks Thanks for checking this out!!

**Comment by robvelor** [comment]
@jderose9 @charlesg99 I have been trying it out for over a week now with no issues, thanks for the fix guys!

**Comment by vageliskatsiotis** [comment]
Still experiencing this issue in Arch.  Gnome Shell 3.36,WM: Mutter
